### PR TITLE
Update @hapi/joi to version 16

### PR DESCRIPTION
**Pull request by CanerSezgin**
Issue: https://github.com/opencollective/opencollective/issues/2602
All tests were passed.

In **src/config.ts** file, line **57** _.requiredKeys(['collective'])_ had been used for making only collective required. However in line **44** there is already required() statement. So I thought, line **57** is redundant. 

Also, in version 16, **requiredKeys** has been deprecated. Instead of this method, **any.fork()** might be used. 

`.fork(['collective'], (schema) => schema.required())`

### CODE REVIEW

**Review by znarf — COMMENTED** [review]
I don't see how it's working as expected without the `.fork(['collective'], (schema) => schema.required())` modification. Shouldn't that be included?

File: src/validate.ts [hunk 1]
```diff
@@ -29,26 +28,21 @@ validator.post(
...
-          const message = err.details
-            .map(
-              (detail: { path: string[]; message: string }) =>
-                `- ${detail.path.join('.')}: ${detail.message}`,
-            )
-            .join('\n')
-
-          res.status(400).send(message)
-        })
+      await configSchema.validateAsync(config, { abortEarly: false })
```
Comment (znarf) [inline src/validate.ts:31]: What is the purpose of this change? Just cleaner code using `async`/`await`? Or was it somehow required for the update?

Comment (CanerSezgin) [reply src/validate.ts:31]: You can use again **then/catch** blocks but for the sake of simplicity and keep the structure consistent (because everywhere **async/await** had been used) I have converted this into **async/await**. By benefiting from **isJoi** property of Joi Error object while seperating errors that will catch. 

Here Then/Catch block solution (also tests passed)
```
      configSchema.validateAsync(config, { abortEarly: false })
        .then(() => {
          res.status(200).send('Valid configuration!')
        })
        .catch(err => {
          /* Compose a report */
          const message = err.details
            .map(
              (detail: { path: string[]; message: string }) =>
                `- ${detail.path.join('.')}: ${detail.message}`,
            )
            .join('\n')

          res.status(400).send(message)
        })
```

**Comment by CanerSezgin** [comment]
> I don't see how it's working as expected without the `.fork(['collective'], (schema) => schema.required())` modification. Shouldn't that be included?

requiredKeys() is used for setting the specified children to required
Here is the example from docs v15 

> const schema = Joi.object().keys({ a: { b: Joi.number() }, c: { d: Joi.string() } });
> const requiredSchema = schema.requiredKeys('', 'a.b', 'c', 'c.d');

I thought, we have already defined required for collective 

>     collective: joi
>       .string()
>       .regex(/^[\w\-]+$/, 'Use your collective slug.')
>       .required(),

So no need to use. Why do you think that we need to use it ?

**Comment by znarf** [comment]
@CanerSezgin I'll check in details but in my first attempt at fixing this issue, this didn't work as expected. Maybe the new error catching changed the behavior. I'll come back with an example.

**Comment by CanerSezgin** [comment]
@znarf  Okay, let me know. I can fix quickly whatever you need. Also writing more tests for extreme cases would be helpful for now and the future.

**Review by znarf — APPROVED** [review]
Ok, so I could not find what I saw in my previous attempts. 

It's great, will merge.
